src/components/PostDetails.tsx

```diff
@@ -2,105 +1,139 @@
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { addComment, deleteComment, getCommentsByPost } from '../api/comments';
+
+type Props = {
+  post: Post | null;
+};
+
+export const PostDetails: React.FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isAddingComment, setIsAddingComment] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  useEffect(() => {
+    if (!post) {
+      return;
+    }
+
+    const loadComments = async () => {
+      setIsLoading(true);
+      setError(false);
+
+      try {
+        const commentsFromServer = await getCommentsByPost(post.id);
+
+        setComments(commentsFromServer);
+      } catch {
+        setError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    loadComments();
+  }, [post]);
+
+  useEffect(() => {
+    setIsAddingComment(false);
+  }, [post]);
+
+  const handleAddComment = useCallback(
+    async (commentData: Omit<Comment, 'id'>) => {
+      try {
+        const newComment = await addComment(commentData);
+
+        setComments([...(comments || []), newComment]);
+      } catch {
+        setError(true);
+      }
+    },
+    [comments],
+  );
+
+  const handleDeleteComment = useCallback(
+    async (id: number) => {
```

Comment:
Consider following the naming conventions for methods as described here: https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364. A better name for this method could be `handleCommentDeletion`.